reimplementation.py

```diff
@@ -245,64 +369,114 @@ def convert_prob_per_mixture_to_marginal_per_class(prob, labels_in_class):
     epsilon = 10 ** -MAX_LR
     res_prob = np.where(res_prob > 1 - epsilon, 1 - epsilon, res_prob)
     res_prob = np.where(res_prob < epsilon, epsilon, res_prob)
+
     return res_prob
 
 
-def read_mixture_data(n_single_cell_types_no_penile, binarize=True):
+def read_mixture_data(n_single_cell_types_no_penile, n_features, classes_map,
+                      binarize=True):
     """
-    reads in the experimental mixture data that is used as test data
-    :param n_single_cell_types_no_penile: int: number of single cell types excluding penile skine
+    Reads in the experimental mixture data that is used as test data.
+
+    Note that the samples are saved in a separate numpy array
+    based on their replicate indices. As the size differs per
+    sample, the array is filled up with zeros to get the correct
+    dimension.
+
+    :param n_single_cell_types_no_penile: int: number of single cell types
+        excluding penile skine
     :param binarize: bool: whether to binarize values
-    :return: N_samples x N_markers array of measurements NB only one replicate per sample,
-                N_samples iterable of mixture class labels - corresponds to the labels used in data augmentation,
-                N_samples x N_single_cell_type n_hot encoding of the labels NB in in single cell type space!
+    :return: N_samples x N_markers array of measurements NB only one replicate per
+                    sample,
+                N_samples iterable of mixture class labels - corresponds to the labels
+                    used in data augmentation,
+                N_samples x N_single_cell_type n_hot encoding of the labels NB in
+                    in single cell type space!
                 dict: mixture name -> list of int single cell type labels
                 dict: mixture class label -> mixture name
     """
     # read test data
-    df = read_df('Datasets/Dataset_mixtures.xlsx', binarize)
-    # restructure data
-    # TODO this is one per replicate currently! - are there replications?
-    test_labels = np.array(df.index)
+    df, rv = read_df('Datasets/Dataset_mixtures_rv.xlsx', binarize)
```

Comment:
deze code is praktisch hetzelfde als het inlezen van de single cell data. kan het gecombineerd worden?